seerep-hdf5/seerep-hdf5-fb/include/seerep-hdf5-fb/hdf5-fb-pointcloud.h

```diff
@@ -0,0 +1,282 @@
+#ifndef SEEREP_HDF5_FB_HDF5_FB_POINT_CLOUD_H_
+#define SEEREP_HDF5_FB_HDF5_FB_POINT_CLOUD_H_
+
+// highfive
+#include <highfive/H5File.hpp>
+
+// seerep-hdf5
+#include <seerep-hdf5-core/hdf5-core-point-cloud.h>
+
+#include "seerep-hdf5-fb/hdf5-fb-general.h"
+#include "seerep-hdf5-fb/hdf5-fb-point-cloud2-iterator.h"
+
+// seerep-msgs
+#include <seerep-msgs/point_cloud_2_generated.h>
+
+// seerep-com
+#include <seerep-com/point_cloud_service.grpc.fb.h>
+
+// std
+#include <boost/geometry.hpp>
+#include <optional>
+
+namespace seerep_hdf5_fb
+{
+class Hdf5FbPointCloud : public Hdf5FbGeneral
+{
+public:
+  /**
+   * @brief Construct a new Hdf5FbPointCloud object
+   *
+   * @param file shared pointer to the hdf5 file of the point cloud
+   * @param write_mtx shared pointer to a mutex
+   */
+  Hdf5FbPointCloud(std::shared_ptr<HighFive::File>& file, std::shared_ptr<std::mutex>& write_mtx);
+
+  /**
+   * @brief Method for writing a flatbuffers PointCloud2 message to hdf5
+   *
+   * @param uuid the uuid of the point cloud
+   * @param pointcloud2 the received PointCloud2 message
+   * @param boundingBox reference to an vector which get's the computed boundingBox during write of x,y,z
+   */
+  void writePointCloud2(const std::string& uuid, const seerep::fb::PointCloud2& pointcloud2,
+                        std::vector<float>& boundingBox);
+
+  /**
+   * @brief Method for reading a flatbuffers PointCloud2 message from hdf5
+   *
+   * @param uuid the uuid of the point cloud
+   * @param withoutData omit the data field
+   * @return std::optional<flatbuffers::grpc::Message<seerep::fb::PointCloud2>> returns the
+   */
+  std::optional<flatbuffers::grpc::Message<seerep::fb::PointCloud2>> readPointCloud2(const std::string& uuid,
+                                                                                     const bool withoutData = false);
+
+private:
+  /**
+   * @brief struct to store information about the PointCloud2
+   *
+   */
+  struct CloudInfo
+  {
+    bool has_points = false;
+    bool has_rgb = false;
+    bool has_rgba = false;
+    bool has_normals = false;
+  };
+
+  /**
+   * @brief Writes information about point fields as attributes to hdf5
+   *
+   * @param object HighFive object to write to (representing a data set or data group)
+   * @param pointFields flatbuffers vector of the point fields to write
+   */
+  template <typename T>
+  void writePointFieldAttributes(HighFive::AnnotateTraits<T>& object,
+                                 const flatbuffers::Vector<flatbuffers::Offset<seerep::fb::PointField>>& pointFields)
```

Comment:
Here also pointer and check if pointer not null. Should happen, because then it's no pointcloud at all, but anyway